crates/editor/src/editor.rs

```diff
@@ -718,6 +709,8 @@ pub struct Editor {
     edit_prediction_preview: EditPredictionPreview,
     edit_prediction_cursor_on_leading_whitespace: bool,
     edit_prediction_requires_modifier_in_leading_space: bool,
+    /// Shows prediction even if disabled for the current buffer
+    user_requested_edit_prediction: bool,
```

Comment:
I would think that we'd be able to solve this in a less stateful way, without adding any fields to `Editor`. Further comments below.